### PR TITLE
Added instruction width to ProDe status bar and misc. fixes

### PR DESCRIPTION
**Pull request by tervoa**
- Added an instruction width indicator to ProDe status bar
- Fixed a bug where pasting a bus to a machine without a bus of the same name would crash ProDe
- Clarified some points in the TCE Tour section of the manual
- Indentation fixes to fit surrounding code

### CODE REVIEW

File: tce/src/applibs/wxToolkit/MachineWindow/MachineCanvas.cc [hunk 1]
```diff
@@ -284,6 +289,21 @@ MachineCanvas::updateMachine() {
     assert(root_ != NULL);
     root_->setContents(contents);
     contents->figure()->setOptions(&options_);
+
+    if (parent_ != NULL){
```
Comment (pjaaskel) [inline tce/src/applibs/wxToolkit/MachineWindow/MachineCanvas.cc]: Space after )

File: tce/src/applibs/wxToolkit/MachineWindow/MachineCanvas.cc [hunk 2]
```diff
@@ -284,6 +289,21 @@ MachineCanvas::updateMachine() {
...
+
+        wxString text(_T(""));
+        try {
+            BinaryEncoding* bem = BEMGenerator(*machine_).generate();
```
Comment (pjaaskel) [inline tce/src/applibs/wxToolkit/MachineWindow/MachineCanvas.cc]: Does 'bem' leak? delete?

File: tce/src/base/bem/BEMTester.cc [hunk 3]
```diff
@@ -73,38 +73,37 @@ BEMTester::canAddComponentEncoding(
...
-	    BridgeEncoding& existingEnc = sField->bridgeEncoding(i);
-            int alignment = calculateAlignment(
-                encodingWidth, existingEnc.width(), field);
-	    int commonBits =
-		commonBitCount(
-		    encoding, extraBits, existingEnc.encoding(),
-		    existingEnc.extraBits(), alignment);
-	    if (commonBits == static_cast<int>(
-                    MathTools::bitLength(encoding)) + 
+    int bridgeEncodings = sField->bridgeEncodingCount();
```
Comment (pjaaskel) [inline tce/src/base/bem/BEMTester.cc]: Is the indentation correct? Remember to use spaces not tabs.

File: tce/src/base/bem/BEMTester.cc [hunk 4]
```diff
@@ -167,38 +166,44 @@ BEMTester::canAddPortEncoding(
...
-	    return false;
-	}
+        RFPortCode& code = table.rfPortCode(i);
+
+        int commonBits = commonBitCount(
+            encoding, extraBits, code.encoding(), code.extraBits(),
+            code.encodingWidth() - encodingWidth);
+
+        if (commonBits == encodingWidth ||
+                commonBits == code.encodingWidth()) {
```
Comment (pjaaskel) [inline tce/src/base/bem/BEMTester.cc]: check indent

File: tce/src/base/bem/BEMTester.cc [hunk 5]
```diff
@@ -167,38 +166,44 @@ BEMTester::canAddPortEncoding(
...
-	    return false;
-	}
+        IUPortCode& code = table.iuPortCode(i);
+
+        int commonBits = commonBitCount(
+            encoding, extraBits, code.encoding(), code.extraBits(),
+            code.encodingWidth() - encodingWidth);
+        
+        if (commonBits == encodingWidth ||
+                commonBits == code.encodingWidth()) {
```
Comment (pjaaskel) [inline tce/src/base/bem/BEMTester.cc]: check indent

File: tce/src/procgen/ProDe/PasteComponentCmd.cc [hunk 6]
```diff
@@ -145,19 +145,29 @@ PasteComponentCmd::Do() {
...
-        paste(*machine, copiedBus, navigator);
-
-        // also copy the guards from the original Bus 
-        assert(copiedBus->guardCount() == 0);
-        for (int i = 0; i < original.guardCount(); ++i) {
-            original.guard(i)->copyTo(*copiedBus);
+        // cannot copy guards from one machine into another
+        // plus additional check if oroginal bus has been deleted
+        if (clipboard->sourceMachine() == machine && 
+                navigator.hasItem(copiedBus->name())) {
```
Comment (pjaaskel) [inline tce/src/procgen/ProDe/PasteComponentCmd.cc]: Check indent.

**Comment by pjaaskel** [comment]
Please check a couple of style issues and a possible mem leak.

**Comment by tervoa** [comment]
Indent checked, conforms to style guide. Old versions had tabs, which were replaced with 8 spaces each.

**Comment by pjaaskel** [comment]
Alright. Yes there are some \t in the UI code.

File: tce/src/base/bem/BEMTester.cc [hunk 7]
```diff
@@ -73,38 +73,37 @@ BEMTester::canAddComponentEncoding(
...
-                    MathTools::bitLength(encoding)) + 
+    int bridgeEncodings = sField->bridgeEncodingCount();
+    for (int i = 0; i < bridgeEncodings; i++) {
+        BridgeEncoding& existingEnc = sField->bridgeEncoding(i);
+        int alignment = calculateAlignment(
+            encodingWidth, existingEnc.width(), field);
+        int commonBits =
+            commonBitCount(
+              encoding, extraBits, existingEnc.encoding(),
+              existingEnc.extraBits(), alignment);
```
Comment (linehill) [inline tce/src/base/bem/BEMTester.cc]: Misindentation here. Arguments of commonBitCount() call does not have indentation of 4 spaces but 2 instead.

File: tce/src/applibs/wxToolkit/MachineWindow/MachineCanvas.cc [hunk 8]
```diff
@@ -284,6 +289,22 @@ MachineCanvas::updateMachine() {
     assert(root_ != NULL);
     root_->setContents(contents);
     contents->figure()->setOptions(&options_);
+
+    if (parent_ != NULL) {
+
+        wxString text(_T(""));
+        BinaryEncoding* bem(NULL);
```
Comment (linehill) [inline tce/src/applibs/wxToolkit/MachineWindow/MachineCanvas.cc]: Prefer nullptr over the old NULL macro.

File: tce/src/base/bem/BEMTester.cc [hunk 9]
```diff
@@ -73,38 +73,37 @@ BEMTester::canAddComponentEncoding(
...
-            + static_cast<int>(extraBits) || 
-            commonBits == existingEnc.socketIDWidth()) {
-	    return false;
-	}
+        int commonBits = commonBitCount(
+            encoding, extraBits, existingEnc.encoding(),
+            existingEnc.extraBits(), alignment);
+        if (commonBits == static_cast<int>(MathTools::bitLength(encoding))
+                + static_cast<int>(extraBits) || 
+                commonBits == existingEnc.socketIDWidth()) {
```
Comment (linehill) [inline tce/src/base/bem/BEMTester.cc]: `commonBits == existingEnc.socketIDWidth()` should be on same indentation level as the first `commonBits == static...`.

File: tce/src/procgen/ProDe/PasteComponentCmd.cc (hunk 6 above)
Comment (linehill) [inline tce/src/procgen/ProDe/PasteComponentCmd.cc]: Seconded.

File: tce/src/base/bem/InstructionField.cc [hunk 10]
```diff
@@ -258,7 +258,7 @@ InstructionField::loadState(const ObjectState* state)
     const string procName = "InstructionField::loadState";
 
     try {
-	setExtraBits(state->intAttribute(OSKEY_EXTRA_BITS));
+    setExtraBits(state->intAttribute(OSKEY_EXTRA_BITS));
```
Comment (linehill) [inline tce/src/base/bem/InstructionField.cc]: Check indentation.

File: tce/src/base/bem/BEMTester.cc (hunk 3 above)
Comment (linehill) [inline tce/src/base/bem/BEMTester.cc]: Indentation needs still fixing here.

File: tce/src/base/bem/BEMTester.cc [hunk 11]
```diff
@@ -73,38 +73,37 @@ BEMTester::canAddComponentEncoding(
...
-            int alignment = calculateAlignment(
-                encodingWidth, existingEnc.width(), field);
-	    int commonBits =
-		commonBitCount(
-		    encoding, extraBits, existingEnc.encoding(),
-		    existingEnc.extraBits(), alignment);
-	    if (commonBits == static_cast<int>(
-                    MathTools::bitLength(encoding)) + 
+    int bridgeEncodings = sField->bridgeEncodingCount();
+    for (int i = 0; i < bridgeEncodings; i++) {
```
Comment (linehill) [inline tce/src/base/bem/BEMTester.cc]: Check indentation.

**Pull request merged**